lang/cs/Org.Apache.REEF.Wake/Remote/AuxillaryStreamingFunctions.cs

```diff
@@ -0,0 +1,275 @@
+﻿using System;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Org.Apache.REEF.Wake.Remote
+{
+    /// <summary>
+    /// Static class containing functions to read and write some basic types from/to stream
+    /// </summary>
+    public static class AuxillaryStreamingFunctions
```

Comment:
Would it simplify the code of this was a wrapper around `Stream`? Have a look at [Java's DataOutput](http://docs.oracle.com/javase/7/docs/api/java/io/DataOutput.html) for an idea of what I mean.

@beysims, @jwang98052: Is there something like it in the C# library already?